src/hotspot/share/cds/cds_globals.hpp

```diff
@@ -94,6 +94,21 @@
            "(2) always map at preferred address, and if unsuccessful, "     \
            "do not map the archive")                                        \
            range(0, 2)                                                      \
+                                                                            \
+  /*========== New "AOT" flags =========================================*/  \
+  /* These are aliases of -Xshare:dump, -XX:SharedArchiveFile=..., etc. */  \
+  /* See CDSConfig::check_flag_aliases().                               */  \
+                                                                            \
+  product(ccstr, AOTMode, nullptr,                                          \
```

Comment:
you can replace nullptr with "auto" as that's the default, that would simplify some of the FLAG_IS_DEFAULT(AOTMode) || (strcmp(AOTMode, "auto") == 0 checks.  So you would now check strcmp(AOTMode, "auto") == 0 which I guess is slightly less performant (as FLAG_IS_DEFAULT is a cheap early out), however the current implementation means that the definition of the default value can spread across the codebase vs only being specified in cds_globals.h

Reply:
If AOTMode has the "auto" value by default, that will be inconsistent if the JVM is started with old parameters like `-Xshare:dump` and `-Xshare:off`. I think it's better to leave it as null to avoid confusions (especially when running inside gdb, etc).